diff --git a/src/ng_package/index.bzl b/src/ng_package/index.bzl
index 5d0bf37..7b92a43 100644
--- a/src/ng_package/index.bzl
+++ b/src/ng_package/index.bzl
@@ -6,7 +6,8 @@ def ng_package(
         name,
         nested_packages = [],
         substitutions = {},
+        stamp = False,
         tags = [],
         replace_prefixes = {},
         data = [],
         **kwargs):
@@ -18,5 +19,6 @@ def ng_package(
     text_replace(
         name = "%s_apf_substituted" % name,
         srcs = [":%s_apf" % name],
         substitutions = substitutions,
+        stamp = stamp,
     )
diff --git a/src/ng_package/text_replace/index.bzl b/src/ng_package/text_replace/index.bzl
index 67dd7bc..3f8c084 100644
--- a/src/ng_package/text_replace/index.bzl
+++ b/src/ng_package/text_replace/index.bzl
@@ -4,7 +4,5 @@ def _text_replace_impl(ctx):
-
-    inputs = [
-        ctx.version_file,
-        ctx.info_file,
-    ] + ctx.files.srcs
-
+
+    status_files = [ctx.version_file, ctx.info_file] if ctx.attr.stamp else []
+    inputs = status_files + ctx.files.srcs
+
     args = ctx.actions.args()
@@ -19,2 +17,2 @@ def _text_replace_impl(ctx):
     # The path to the volite status file for substitution of values from the file
-    args.add(ctx.version_file.path)
+    args.add(ctx.version_file.path if ctx.attr.stamp else "")
@@ -22,2 +20,2 @@ def _text_replace_impl(ctx):
     # The path to the stable status file for substitution of values from the file
-    args.add(ctx.info_file.path)
+    args.add(ctx.info_file.path if ctx.attr.stamp else "")
@@ -59,6 +57,10 @@ text_replace = rule(
         "substitutions": attr.string_dict(
             doc = """Key-value pairs which are replaced in all the files provided.""",
             mandatory = True,
         ),
+        "stamp": attr.bool(
+            doc = "Whether to make substitutions from Bazel workspace status files.",
+            default = False,
+        ),
         "_text_replace": attr.label(
             doc = "The binary used to process the provided files and apply text substitutions",
diff --git a/src/worker/worker.mts b/src/worker/worker.mts
index 6a729f9..9b65863 100644
--- a/src/worker/worker.mts
+++ b/src/worker/worker.mts
@@ -13,2 +13,2 @@ export async function bootWorker(optionalAngular?: OptionalAngular) {
   if (!worker.isPersistentWorker(process.argv)) {
-    const isRemoteExecution = process.cwd().startsWith('/b/f/w/');
+    const isRemoteExecution = process.cwd().startsWith('/b/f/w/') || process.cwd().startsWith('/var/lib/namespace-bazel/');
